src/pages/dataset-list-view/dataset-list-view.js

```diff
@@ -1,232 +1,137 @@
-            />
-            <FacetSidebar
-              availableFacets={this.props.searchMetadata.facets}
-              appliedFacets={this.props.searchParams.facets}
-              clickHandler={(facetName, facetValue) =>
-                this.onFacetClick(facetName, facetValue)
-              }
+  const renderResultsCountText = () => {
+    const resultCountText = isSearchLoading
+          ? ""
+          : (`${searchMetadata.totalDatasets || "No"} datasets found`)
+    const resultCountQueryText = searchText
+          ? ` for "${searchText}"`
+          : ""
+    return <div className="result-count">{`${resultCountText}${resultCountQueryText}`}</div>
+  }
+
+  const token = sessionStorage.getItem("api-token")
+
+  if (error) {
+    return (
+      <ErrorComponent
+        errorText={
+          "We were unable to fetch the datasets, please refresh the page to try again"
+        }
+      />
+    )
+  } else {
+    return (
+      <dataset-list-view>
+        <div className="left-section">
+          <LoginZone token={token} />
+          <Checkbox
+            clickHandler={toggleApiAccessible}
+            text="API Accessible"
+            selected={apiAccessible}
+          />
+          <FacetSidebar
+            availableFacets={searchMetadata.facets}
+            appliedFacets={facets}
+            clickHandler={toggleFacet}
+          />
+        </div>
+        <div className="right-section">
+          <Search
+            className="search"
+            defaultText={searchText}
+            placeholder="Search datasets"
+            callback={updateSearchText}
+          />
+          <div className="list-header">
+            {renderResultsCountText()}
+            <Select
+              className="sortOrder-select"
+              label="order by"
+              options={createSortOptions()}
+              selectChangeCallback={updateSortOrder}
             />
           </div>
-          <div className="right-section">
-            <Search
-              className="search"
-              defaultText={this.props.searchParams.query}
-              placeholder="Search datasets"
-              callback={searchCriteria => this.onSearchChange(searchCriteria)}
-            />
-            <div className="list-header">
-              <div className="result-count">{`${resultCountText}${resultCountQueryText}`}</div>
-              <Select
-                className="sort-select"
-                label="order by"
-                options={this.createSortOptions}
-                selectChangeCallback={sort => this.onSortChange(sort)}
-              />
-            </div>
-            {this.renderLoading()}
-            <Paginator
-              className="paginator"
-              numberOfPages={this.props.numberOfPages}
-              currentPage={this.props.pageNumber}
-              pageChangeCallback={page => this.onPageChange(page)}
-            />
-          </div>
-        </dataset-list-view>
-      );
-    }
```

Comment:
![Alt Text](https://media.giphy.com/media/jDOuYPr7NXpoQ/giphy.gif)

Reply:
My god. We can gif here?